compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -569,7 +569,12 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     });
                 }
 
-                let path = if trees.is_empty() && !prefix.segments.is_empty() {
+                // Condition should match `build_reduced_graph_for_use_tree`.
```

Comment:
The segment types are, unfortunately, different between here and resolve, so it cannot be just shared.